### PR TITLE
[ADF-1980] fix search component styles

### PR DESCRIPTION
**Pull request by DenisVuyka**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [x] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)

**What is the new behaviour?**

- removes the hack from the demo shell styles (app layout)
- fixes overflow issue on the component level
- fixes crashes related to "OnDestroy" events

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: demo-shell/src/app/components/search/search-bar.component.ts [hunk 1]
```diff
@@ -27,23 +26,7 @@ import { AuthenticationService } from '@alfresco/adf-core';
 })
 export class SearchBarComponent {
 
-    @Input()
-    expandable: boolean = true;
```
Comment (eromano) [inline demo-shell/src/app/components/search/search-bar.component.ts]: can you remove it also from the documentation?

File: demo-shell/src/app/components/search/search-bar.component.ts [hunk 2]
```diff
@@ -27,23 +26,7 @@ import { AuthenticationService } from '@alfresco/adf-core';
...
-
-    @Output()
-    expand = new EventEmitter();
```
Comment (eromano) [inline demo-shell/src/app/components/search/search-bar.component.ts]: also expand

File: demo-shell/src/app/components/search/search-bar.component.ts (hunk 1 above)
Comment (DenisVuyka) [reply demo-shell/src/app/components/search/search-bar.component.ts]: that's the demo shell wrapper component, cleaned up some properties left from previous refactorings

File: demo-shell/src/app/components/search/search-bar.component.ts (hunk 2 above)
Comment (DenisVuyka) [reply demo-shell/src/app/components/search/search-bar.component.ts]: not used in components, just an old demo shell code

**Pull request merged**